### PR TITLE
Kucoin stop price

### PR DESCRIPTION
**Pull request by samgermain**
```
% kucoin createOrder ADA/USDT limit sell 10 0.92 '{"stopPrice": 0.92}'
2022-04-15T05:22:59.395Z
Node.js: v14.17.0
CCXT v1.79.17
kucoin.createOrder (ADA/USDT, limit, sell, 10, 0.92, [object Object])
2022-04-15T05:23:00.992Z iteration 0 passed in 262 ms

{
  id: 'vs86koip04qlmsrn000h4ncq',
  clientOrderId: '...',
  info: { orderId: 'vs86koip04qlmsrn000h4ncq' },
  timestamp: 1650000180992,
  datetime: '2022-04-15T05:23:00.992Z',
  lastTradeTimestamp: undefined,
  symbol: 'ADA/USDT',
  type: 'limit',
  side: 'sell',
  price: 0.92,
  amount: 10,
  cost: undefined,
  average: undefined,
  filled: undefined,
  remaining: undefined,
  status: undefined,
  fee: undefined,
  trades: undefined
}

% kucoin fetchOrder vs86koip05b4tii0000novl5 undefined '{"stop": true}'
2022-04-15T06:13:05.187Z
Node.js: v14.17.0
CCXT v1.79.17
kucoin.fetchOrder (vs86koip05b4tii0000novl5, , [object Object])
2022-04-15T06:13:07.164Z iteration 0 passed in 252 ms

{
  id: 'vs86koip05b4tii0000novl5',
  clientOrderId: '...',
  symbol: 'ADA/USDT',
  type: 'limit',
  timeInForce: 'GTC',
  postOnly: false,
  side: 'sell',
  amount: 10,
  price: 0.92,
  stopPrice: 0.92,
  cost: undefined,
  filled: undefined,
  remaining: undefined,
  timestamp: 1650000214239,
  datetime: '2022-04-15T05:23:34.239Z',
  fee: { currency: 'USDT', cost: undefined },
  status: 'canceled',
  info: {
    id: 'vs86koip05b4tii0000novl5',
    symbol: 'ADA-USDT',
    userId: '...',
    status: 'NEW',
    type: 'limit',
    side: 'sell',
    price: '0.92000000000000000000',
    size: '10.00000000000000000000',
    funds: null,
    stp: null,
    timeInForce: 'GTC',
    cancelAfter: -1,
    postOnly: false,
    hidden: false,
    iceberg: false,
    visibleSize: null,
    channel: 'API',
    clientOid: '...',
    remark: null,
    tags: null,
    orderTime: 1650000214238000000,
    domainId: 'kucoin',
    tradeSource: 'USER',
    tradeType: 'TRADE',
    feeCurrency: 'USDT',
    takerFeeRate: '0.00100000000000000000',
    makerFeeRate: '0.00100000000000000000',
    createdAt: 1650000214239,
    stop: 'loss',
    stopTriggerTime: null,
    stopPrice: '0.92000000000000000000'
  },
  lastTradeTimestamp: undefined,
  average: undefined,
  trades: [],
  fees: [ { currency: 'USDT', cost: undefined } ]
}

% kucoin fetchOrder undefined undefined '{"stop": true, "clientOid": "..."}'
kucoin.fetchOrder (, , [object Object])
2022-04-15T06:14:35.485Z iteration 0 passed in 230 ms

{
  id: undefined,
  clientOrderId: undefined,
  symbol: undefined,
  type: undefined,
  timeInForce: undefined,
  postOnly: undefined,
  side: undefined,
  amount: undefined,
  price: undefined,
  stopPrice: undefined,
  cost: undefined,
  filled: undefined,
  remaining: undefined,
  timestamp: undefined,
  datetime: undefined,
  fee: { currency: undefined, cost: undefined },
  status: 'closed',
  info: [
    {
      id: 'vs86koip04qlmsrn000h4ncq',
      symbol: 'ADA-USDT',
      userId: '...',
      status: 'NEW',
      type: 'limit',
      side: 'sell',
      price: '0.92000000000000000000',
      size: '10.00000000000000000000',
      funds: null,
      stp: null,
      timeInForce: 'GTC',
      cancelAfter: -1,
      postOnly: false,
      hidden: false,
      iceberg: false,
      visibleSize: null,
      channel: 'API',
      clientOid: '...',
      remark: null,
      tags: null,
      orderTime: 1650000181132000000,
      domainId: 'kucoin',
      tradeSource: 'USER',
      tradeType: 'TRADE',
      feeCurrency: 'USDT',
      takerFeeRate: '0.00100000000000000000',
      makerFeeRate: '0.00100000000000000000',
      createdAt: 1650000181132,
      stop: 'loss',
      stopTriggerTime: null,
      stopPrice: '0.92000000000000000000'
    }
  ],
  lastTradeTimestamp: undefined,
  average: undefined,
  trades: [],
  fees: [ { currency: undefined, cost: undefined } ]
}

% kucoin cancelOrder undefined undefined '{"clientOid": "...", "stop": true}'
kucoin.cancelOrder (, , [object Object])
2022-04-15T06:18:51.715Z iteration 0 passed in 244 ms

{
  code: '200000',
  data: {
    cancelledOrderId: 'vs86koip04qlmsrn000h4ncq',
    clientOid: '...'
  }
}

% kucoin fetchOpenOrders undefined undefined undefined '{"stop": true}'
2022-04-15T06:20:05.597Z
Node.js: v14.17.0
CCXT v1.79.17
kucoin.fetchOpenOrders (, , , [object Object])
2022-04-15T06:20:07.650Z iteration 0 passed in 244 ms

                      id |                        clientOrderId |   symbol |  type | timeInForce | postOnly | side | amount | price | stopPrice | cost | filled | remaining |     timestamp |                 datetime |                 fee |   status | lastTradeTimestamp | average | trades |                  fees
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
vs86koip05b4tii0000novl5 | ... | ADA/USDT | limit |         GTC |    false | sell |     10 |  0.92 |      0.92 |      |        |           | 1650000214239 | 2022-04-15T05:23:34.239Z | {"currency":"USDT"} | canceled |                    |         |     [] | [{"currency":"USDT"}]
1 objects

% kucoin cancelOrder vs86koip05b4tii0000novl5 undefined '{"stop": true}'
2022-04-15T06:21:05.190Z
Node.js: v14.17.0
CCXT v1.79.17
kucoin.cancelOrder (vs86koip05b4tii0000novl5, , [object Object])
2022-04-15T06:21:06.600Z iteration 0 passed in 247 ms

{
  code: '200000',
  data: { cancelledOrderIds: [ 'vs86koip05b4tii0000novl5' ] }
}

% kucoin cancelAllOrders undefined '{"stop": true}'
2022-04-15T06:22:45.868Z
Node.js: v14.17.0
CCXT v1.79.17
kucoin.cancelAllOrders (, [object Object])
2022-04-15T06:22:47.040Z iteration 0 passed in 245 ms

{
  code: '200000',
  data: {
    cancelledOrderIds: [ 'vs86koip1s1sci4e000tb3op', 'vs86koip1s7uve3f000jml1r' ]
  }
}
```

### CODE REVIEW

**Comment by samgermain** [comment]
For some reason `fetchOrder` with  clientOid for stopPrice parses all the fields as undefined, even though the exchange response looks the same

**Comment by kroitor** [comment]
> For some reason `fetchOrder` with clientOid for stopPrice parses all the fields as undefined, even though the exchange response looks the same

That can happen if `parseOrder` is fed with an empty object.

**Comment by samgermain** [comment]
> > For some reason `fetchOrder` with clientOid for stopPrice parses all the fields as undefined, even though the exchange response looks the same
> 
> That can happen if `parseOrder` is fed with an empty object.

I figured it out

File: js/kucoin.js [hunk 1]
```diff
@@ -1102,6 +1102,34 @@ module.exports = class kucoin extends Exchange {
...
+         * @method
+         * @name kucoin#createOrder
+         * @description Create an order on the exchange
+         * @param {str} symbol Unified CCXT market symbol
+         * @param {str} type "limit" or "market"
+         * @param {str} side "buy" or "sell"
+         * @param {float} amount the amount of currency to trade
+         * @param {float} price *ignored in "market" orders* the price at which the order is to be fullfilled at in units of the quote currency
+         * @param {dict} params  Extra parameters specific to the exchange API endpoint
+         * @param {float} params.leverage Leverage size of the order
```
Comment (kroitor) [inline js/kucoin.js:1115]: We should not use the js-specific syntax in the docs, so we must replace `params.leverage` with `params['leverage']` and all other similar cases everywhere to keep it uniform with the other languages.

Tbh, I don't really like the idea of moving these params to the docstrings, cause that makes the code less readable.

Comment (samgermain) [reply js/kucoin.js]: Leave this as is and I'll add a transpilation. This is more readable in my opinion, and it's available to the user, without this, the only way the user can really know how to set leverage on kucoin is by checking kucoins documentation, or searching CCXT issues or examples.

How is this appearing in you editor? The highlighting alone makes this much more readable for me

Comment (kroitor) [reply js/kucoin.js]: Let's still do _replace params.leverage with params['leverage']_ for it to work with all languages.

Comment (samgermain) [reply js/kucoin.js]: > Let's still do _replace params.leverage with params['leverage']_ for it to work with all languages (without changing the transpiler specifically for this purpose).

`params.leverage` is proper jsdoc and it's needed to auto create docs, it's already transpiled in PHP, we just need the python transpilation

**Pull request merged**